torch/_inductor/ir.py

```diff
@@ -342,7 +346,23 @@ def get_device_type(
 
 
 def is_triton(x: Union[IRNode, torch.device, None, str]) -> bool:
-    return is_gpu(get_device_type(x))
+    device = get_device_type(x)
+    # Special case cpu and cuda as using the method below
+    # to determine if the scheduler is a triton scheduler subclass
+    # requires instantiating a scheduler for them
+    if device in ["cpu", "cuda"]:
+        if getattr(config, f"{device}_backend") == "triton":
+            return True
+        return False
+    if (
+        device is None
+        or (device_scheduling := get_scheduling_for_device(device)) is None
+    ):
+        return False
+    from .codegen.triton import TritonScheduling
+
+    assert isinstance(device_scheduling, type)
+    return issubclass(device_scheduling, TritonScheduling)
```

Comment:
@kundaMwiza can you describe what this does? why was the previous change (just checking if `getattr(config, f"{device}_backend) == "triton"`) not sufficient?

Reply:
@davidberard98  This is because currently the only devices that have a {device}_backend attribute are CPU and CUDA. Other devices in/out of tree don't have this attribute

Reply:
Additionally, CPU/CUDA define the scheduler via a lambda, which returns the currently active scheduler based on `{device}_backend`. The other devices only have a single scheduler, and so it should be enough to just check that they subclass from `TritonScheduling`.

Reply:
for simplicity, would something like this work?

```python
return is_gpu(get_device_type(x)) or (get_device_type(x) == "cpu" and config.cpu_backend == "triton")
```

or is there another benefit to using the `TritonScheduling` logic?

Reply:
That works for CPU and GPU, but unfortunately not for other third party backends / devices that are using triton by subclassing from `TritonScheduling`

Reply:
ok - seems reasonable to me, I'll wait 1 day to give @jansel and @eellison to take a look and otherwise will stamp tomorrow